### PR TITLE
link: add AttachSkLookup

### PR DESCRIPTION
**Pull request by lmb**
Add a wrapper for SkLookup programs that allows retrieving information about the link.

### CODE REVIEW

**Comment by jsitnicki** [comment]
Looks like nothing in `link/sk_lookup.go` is specific to BPF sk_lookup. BPF flow_dissector would be attached and loaded from pinned link the same way. Have you considered orienting it around BPF link type (`BPF_LINK_TYPE_NETNS`) instead of BPF program type?

**Comment by lmb** [comment]
Good point, I'll do that.

**Comment by lmb** [comment]
Fixed, PTAL.

File: link/link.go [hunk 1]
```diff
@@ -159,3 +190,25 @@ func (l *RawLink) UpdateArgs(opts RawLinkUpdateOptions) error {
...
+	id      uint32
+	prog_id uint32
+}
+
+// Info returns metadata about the link.
+func (l *RawLink) Info() (*RawLinkInfo, error) {
+	var info bpfLinkInfo
+	err := internal.BPFObjGetInfoByFD(l.fd, unsafe.Pointer(&info), unsafe.Sizeof(info))
+	if err != nil {
+		return nil, fmt.Errorf("sk_lookup info: %s", err)
```
Comment (jsitnicki) [inline link/link.go:206]: Error message needs updating.

File: link/syscalls.go [hunk 2]
```diff
@@ -10,6 +10,22 @@ import (
...
+//
+// Equivalent to enum bpf_link_type.
+const (
+	UnspecifiedType Type = iota
+	RawTracepointType
+	TracingType
+	CgroupType
+	IterType
+	NetNsType
+	XDPType
```
Comment (jsitnicki) [inline link/syscalls.go:26]: Is that name unique enough? Why not `…LinkType`?

**Review by jsitnicki — APPROVED** [review]

File: link/link.go (hunk 1 above)
Comment (lmb) [reply link/link.go:206]: Aye

File: link/syscalls.go (hunk 2 above)
Comment (lmb) [reply link/syscalls.go:26]: Because it would read as `link.XDPLinkType` which seems a bit redundant.

Comment (jsitnicki) [reply link/syscalls.go:26]: Makes sense. Didn't notice it's in its own package.

**Pull request merged**